packages/zoe/src/zoeService/types.ts

```diff
@@ -0,0 +1,289 @@
+ * single `invitationIssuer` for the entirety of its lifetime. By
+ * having a reference to Zoe, a user can get the `invitationIssuer`
+ * and thus validate any `invitation` they receive from someone else.
+ *
+ * Zoe has two different facets: the public Zoe service and the
+ * contract facet (ZCF). Each contract instance has a copy of ZCF
+ * within its vat. The contract and ZCF never have direct access to
+ * the users' payments or the Zoe purses.
+ */
+export type ZoeService = {
+  /**
+   * Zoe has a single `invitationIssuer` for the entirety of its
+   * lifetime. By having a reference to Zoe, a user can get the
+   * `invitationIssuer` and thus validate any `invitation` they receive
+   * from someone else. The mint associated with the invitationIssuer
+   * creates the ERTP payments that represent the right to interact with
+   * a smart contract in particular ways.
+   */
+  getInvitationIssuer: GetInvitationIssuer;
+  install: InstallBundle;
+  installBundleID: InstallBundleID;
+  startInstance: import('./utils.js').StartInstance;
+  offer: Offer;
+  getPublicFacet: <I extends Instance>(
+    instance: ERef<I>,
+  ) => Promise<
+    IsAny<I> extends true
+      ? any
+      : I extends import('./utils.js').Instance<
+            infer SF extends ContractStartFunction
+          >
+        ? IsAny<SF> extends true
+          ? unknown
+          : EReturn<SF>['publicFacet']
+        : never
+  >;
+  getIssuers: GetIssuers;
+  getBrands: GetBrands;
+  getTerms: <I extends ERef<Instance>>(
+    instance: I,
+  ) => IsAny<I> extends true
+    ? Promise<any>
+    : I extends ERef<
+          import('./utils.js').Instance<infer SF extends ContractStartFunction>
+        >
+      ? IsAny<SF> extends true
+        ? Promise<unknown>
+        : Promise<StartParams<SF>['terms']>
+      : never;
+  getOfferFilter: (instance: ERef<Instance>) => Promise<string[]>;
+  getInstallationForInstance: GetInstallationForInstance;
+  getInstance: GetInstance;
+  getInstallation: GetInstallation;
+  /**
+   *   Return an object with the instance, installation, description, invitation
+   *   handle, and any custom properties specific to the contract.
+   */
+  getInvitationDetails: GetInvitationDetails;
+  getFeeIssuer: GetFeeIssuer;
+  getConfiguration: GetConfiguration;
+  getBundleIDFromInstallation: GetBundleIDFromInstallation;
+  /**
+   *   Return the pattern (if any) associated with the invitationHandle that a
+   *   proposal is required to match to be accepted by zoe.offer().
+   */
+  getProposalShapeForInvitation: (
+    invitationHandle: InvitationHandle,
+  ) => import('@endo/patterns').Pattern | undefined;
+};
+type GetInvitationIssuer = () => Promise<Issuer<'set', InvitationDetails>>;
+type GetFeeIssuer = () => Promise<Issuer<'nat'>>;
+type GetConfiguration = () => {
+  feeIssuerConfig: FeeIssuerConfig;
+};
+export type GetIssuers = (
+  instance: import('./utils.js').Instance<any>,
+) => Promise<IssuerKeywordRecord>;
```

Comment:
I hope we move these method types to the method definitions.

Except in the case when there's really an interface that's supposed to be implemented in multiple cases.